src/ploomber/spec/taskspec.py

```diff
@@ -200,11 +200,13 @@ def __init__(self,
                  meta,
                  project_root,
                  lazy_import=False,
-                 reload=False):
+                 reload=False,
+                 task_defaults=""):
         self.data = deepcopy(data)
         self.meta = deepcopy(meta)
         self.project_root = project_root
         self.lazy_import = lazy_import
+        self.task_defaults = task_defaults
```

Comment:
and here `self.task_defaults = task_defaults or {}` so if passed None, then it initializes an empty dictionary